reconcile/utils/saasherder.py

```diff
@@ -388,13 +390,41 @@ def _process_template(self, options):
                 return None, None
 
             if "IMAGE_TAG" not in consolidated_parameters:
-                template_parameters = template.get('parameters')
-                if template_parameters is not None:
-                    for template_parameter in template_parameters:
-                        if template_parameter['name'] == 'IMAGE_TAG':
-                            # add IMAGE_TAG only if it is required
-                            image_tag = commit_sha[:hash_length]
-                            consolidated_parameters['IMAGE_TAG'] = image_tag
+                for template_parameter in template.get("parameters", {}):
+                    if template_parameter["name"] == "IMAGE_TAG":
+                        # add IMAGE_TAG only if it is required
+                        image_tag = commit_sha[:hash_length]
+                        consolidated_parameters["IMAGE_TAG"] = image_tag
+
+            # Do this in a separate loop, since it relies on IMAGE_TAG already
+            # being calculated.
+            # This should almost never already be set in
+            # consolidated_parameters, but just in case...
+            if "REPO_DIGEST" not in consolidated_parameters:
+                for template_parameter in template.get("parameters", {}):
+                    if template_parameter["name"] == "REPO_DIGEST":
+                        try:
+                            logging.info("Generating REPO_DIGEST.")
+                            reg_img = consolidated_parameters["REGISTRY_IMG"]
+                            img_tag = consolidated_parameters["IMAGE_TAG"]
+                        except KeyError as e:
+                            logging.error(
+                                f"[{saas_file_name}/{resource_template_name}] "
+                                + f"{html_url}: error generating REPO_DIGEST. "
+                                + f"Is REGISTRY_IMG or IMAGE_TAG missing? "
```

Comment:
IMAGE_TAG will never be missing

Reply:
Based on L392-394 -- and reinforced by the comment on L395 -- it looks like it can be missing if it's not included in the saas parameters or the template parameters. Is there somewhere else that it's magically injected in that case?

Reply:
Ah you are right